From 592fb8305313bfc67131bb6cc169bd9175c0c336 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 8 Sep 2014 23:59:30 +0100
Subject: [PATCH] debugger: Don't skip first char after "This GDB was
 configured as \""

sizeof(needle) includes the \0 terminator, so the first letter of the
target triplet was being skipped too. This caused the architecture to
be mis-detected and (for us on MSYS2) the 32bit GDB to be selected in
64bit Kits."

Change-Id: Id54bc320baf6cc604f2d41fa7ff05aca0756a296
---
 src/plugins/debugger/debuggeritem.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/debugger/debuggeritem.cpp b/src/plugins/debugger/debuggeritem.cpp
index f95d2e3..fe49240 100644
--- a/src/plugins/debugger/debuggeritem.cpp
+++ b/src/plugins/debugger/debuggeritem.cpp
@@ -110,7 +110,7 @@ void DebuggerItem::reinitializeFromFile()
         // or "i686-linux-gnu"
         int pos1 = ba.indexOf(needle);
         if (pos1 != -1) {
-            pos1 += int(sizeof(needle));
+            pos1 += int(sizeof(needle)-1);
             int pos2 = ba.indexOf('"', pos1 + 1);
             QByteArray target = ba.mid(pos1, pos2 - pos1);
             int pos3 = target.indexOf("--target=");
-- 
2.1.0

